arangod/Aql/AqlFunctionFeature.cpp

```diff
@@ -434,6 +434,7 @@ void AqlFunctionFeature::addMiscFunctions() {
   add({"PARSE_IDENTIFIER", ".", flags, &Functions::ParseIdentifier});
   add({"IS_SAME_COLLECTION", ".h,.h", flags, &Functions::IsSameCollection});
   add({"DECODE_REV", ".", flags, &Functions::DecodeRev});
+  add({"SHARD_ID", ".h|.", flags, &Functions::ShardId});
```

Comment:
we must disallow using this function in `AqlAnalyzer`

Reply:
added `FF::CanReadDocuments` to make sure that cannot be used in ANALYZERS